pkg/store/store.go

```diff
@@ -140,9 +141,17 @@ type relationMaps struct {
 
 func (s *atomixStore) Create(ctx context.Context, object *topoapi.Object) error {
 	// If an object is a relation and its ID is empty, build one.
-	if object.ID == "" && object.Type == topoapi.Object_RELATION {
-		relation := object.GetRelation()
-		object.ID = topoapi.RelationID(relation.SrcEntityID, relation.KindID, relation.TgtEntityID)
+	if object.Type == topoapi.Object_RELATION {
+		if object.ID == "" {
+			relation := object.GetRelation()
+			object.ID = topoapi.RelationID(relation.SrcEntityID, relation.KindID, relation.TgtEntityID)
+		}
+		_, srcErr := s.objects.Get(ctx, string(object.GetRelation().SrcEntityID))
+		_, tgtErr := s.objects.Get(ctx, string(object.GetRelation().TgtEntityID))
```

Comment:
As a side-note, this still leaves open possibility that the entities will disappear before we had a chance to create the relation, it's a bit of a race, but we can plug that hole up via handling this in the watch callback later. I'd add a note to that effect here as a TODO.